dvc/utils/__init__.py

```diff
@@ -245,7 +246,24 @@ def load_stage_file_fobj(fobj, path):
         raise StageFileCorruptedError(path)
 
 
+def dvc_walk(top, topdown=True, onerror=None, followlinks=False):
+    """
+    Proxy for `os.walk` directory tree generator.
+    Utilizes DvcIgnoreFilter functionality.
+    """
+    ignore_filter = DvcIgnoreFilter()
```

Comment:
Here you are creating a new filter, that will only be able to account for dvcignores that are present in `top` dir and down below, and will ignore dvcignores specified in parent directory.

Reply:
That could be dangerous, thanks for pointing out!